src/base/file_utils.cpp

```diff
@@ -487,8 +487,15 @@ void move(const std::string& from_path, const std::string& to_path) {
 
   // Rename the file.
 #ifdef _WIN32
-  const auto success =
-      (_wrename(utf8_to_ucs2(from_path).c_str(), utf8_to_ucs2(to_path).c_str()) == 0);
+  std::wstring wto_path = utf8_to_ucs2(to_path);
+  const auto success = (_wrename(utf8_to_ucs2(from_path).c_str(), wto_path.c_str()) == 0);
+  if (success) {
+    // Open the target file for writing without doing any actual writes.
+    // This forces the registration of `to_path` as written by MSBuild FileTracker
+    // - i.e. adds an entry in *.write.*.tlog files.
+    auto handle = CreateFileW(wto_path.c_str(), GENERIC_WRITE, 0, 0, OPEN_ALWAYS, 0, 0);
+    CloseHandle(handle);
+  }
```

Comment:
I'd extract this into a separate function and call it when I need it (not forcing all users of `move` to create & close a file handle every time). `success` should be returned from `move` to callers anyway...

Reply:
Found a simpler solution in https://github.com/shapr3d/buildcache/pull/4/commits/6a83086b8254c0388a647ddf8f8e66ab766fb678